internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,468 @@
+	}
+	if string(gf.DebugData) != d {
+		t.Errorf("ReadFrame(): DebugData: got %q, want %q", string(gf.DebugData), d)
+	}
+	gf.Free()
+}
+
+func (s) TestBridge_ReadFrame_WindowUpdate(t *testing.T) {
+	c := &testConn{}
+	c.rbuf = append(c.rbuf, 0, 0, 4, byte(FrameTypeWindowUpdate), 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	c.rbuf = appendUint32(c.rbuf, 100)
+
+	f := NewFramerBridge(c, c, 0)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	h := fr.Header()
+	checkReadHeader(h, 4, FrameTypeWindowUpdate, 0, 1)
+
+	wf := fr.(*WindowUpdateFrame)
+	if wf.Inc != 100 {
+		t.Errorf("ReadFrame(): Inc: got %d, want %d", wf.Inc, 1)
+	}
+}
+
+func (s) TestBridge_ReadFrame_MetaHeaders(t *testing.T) {
+	fields := []hpack.HeaderField{
+		{Name: "foo", Value: "bar"},
+		{Name: "baz", Value: "qux"},
+	}
+
+	c := &testConn{}
+	enc := hpack.NewEncoder(c)
+	for _, field := range fields {
+		enc.WriteField(field)
+	}
+	half1 := c.wbuf[0 : len(c.wbuf)/2]
+	half2 := c.wbuf[len(c.wbuf)/2:]
+
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half1)), byte(FrameTypeHeaders), 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	// Copy half of the data written by the encoder into the reading buf
+	c.rbuf = append(c.rbuf, half1...)
+
+	// Add Continuation Frame to test merging
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half2)), byte(FrameTypeContinuation), byte(FlagContinuationEndHeaders))
+	c.rbuf = appendUint32(c.rbuf, 1)
+	// Copy data written by the encoder into the reading buf
+	c.rbuf = append(c.rbuf, half2...)
+
+	f := NewFramerBridge(c, c, 0)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	mf, ok := fr.(*MetaHeadersFrame)
+	if !ok {
+		t.Errorf("ReadFrame(): Type: expected MetaHeadersFrame, got %T", fr)
+	}
+	if len(mf.Fields) != 2 {
+		t.Errorf("ReadFrame(): Fields: got %d, want %d", len(mf.Fields), 1)
+	}
+	for i, field := range fields {
+		if field.Name != mf.Fields[i].Name {
+			t.Errorf("ReadFrame(): Fields[%d].Name: got %q, want %q", i, mf.Fields[i].Name, field.Name)
+		}
+		if field.Value != mf.Fields[i].Value {
+			t.Errorf("ReadFrame(): Fields[%d].Value: got %q, want %q", i, mf.Fields[i].Value, field.Value)
+		}
+	}
+
+}
+
+func (s) TestBridge_WriteData(t *testing.T) {
+	c := &testConn{}
+	wantData := "test data"
+	testBuf := [][]byte{[]byte("test"), []byte(" data")}
+	f := NewFramerBridge(c, c, 0)
+	f.WriteData(1, false, testBuf...)
+	if errs := checkWrittenHeader(c.wbuf[0:9], len(wantData), FrameTypeData, 0, 1); len(errs) > 0 {
+		t.Errorf("WriteData():\n%s", errors.Join(errs...))
+	}
+	if string(c.wbuf[9:]) != wantData {
+		t.Errorf("WriteData(): Data: got %q, want %q", string(c.wbuf[9:]), wantData)
+	}
+}
+
+func (s) TestBridge_WriteHeaders(t *testing.T) {
+	tests := []struct {
+		endStream  bool
+		endHeaders bool
+	}{
+		{endStream: false, endHeaders: false},
+		{endStream: true, endHeaders: false},
+		{endStream: false, endHeaders: true},
+		{endStream: true, endHeaders: true},
```

Comment:
Always consider using `t.Run` when writing sub-tests because it has a lot of advantages:
- it asks you to give each sub-test a name and that makes the table more readable
- if one or more tests in the table are failing, `t.Run` makes it possible for you run only subtests of your choice.

Reply:
Done.